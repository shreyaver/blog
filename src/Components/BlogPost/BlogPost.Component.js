import React, { Component } from 'react';
import PostContent from '../PostContent/PostContent.Component.js';
import PostReaction from '../PostReaction/PostReaction.Component.js';
import './BlogPost.Component.css'

class BlogPost extends Component {
    state = {
        liked: this.props.cardData.liked,
        claps: this.props.cardData.claps,
    }
    handleLike = () => {
        this.setState((state) => ({
            liked: !state.liked
        }));
    }
    handleClap = () => {
        this.setState((state) => ({
            claps: state.claps+1
        }));
    }
    render() {
        return (
           <div className = "Blog-post">
            <PostContent cardData={this.props.cardData} />
            <PostReaction claps={this.state.claps} liked={this.state.liked} onLike={this.handleLike}  onClap={this.handleClap}/>
           </div>
        )
    }
}
export default BlogPost;
