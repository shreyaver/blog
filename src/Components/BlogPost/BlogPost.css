.Blog-post {
    width: 30%;
    margin-top: 3%;
    color:grey;
    border: 1px solid rgb(216, 216, 216); 
    min-width: 300px;
}