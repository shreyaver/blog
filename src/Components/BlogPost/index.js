import React, { Component } from 'react';
import PostContent from '../PostContent';
import PostReaction from '../PostReaction';
import './BlogPost.css'

class BlogPost extends Component {
    render() {
        return (
           <div className = "Blog-post">
            <PostContent cardData={this.props.cardData} />
            <PostReaction claps={this.props.cardData.claps} liked={this.props.cardData.liked} />
           </div>
        )
    }
}
export default BlogPost;
