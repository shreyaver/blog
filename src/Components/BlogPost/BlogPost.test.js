import React from 'react';
import renderer from 'react-test-renderer';
import BlogPost from '.';

describe('BlogPost', () => {
  const cardData = {
    "date": "2nd Januray, 2018",
    "readingTime": "2 mins",
    "title": "The future of abstract art and the culture ...",
    "description": "Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...",
    "claps": 10,
    "liked": false,
    "image": "abstract.png"
  }
  it('renders without crashing', () => {
    const blogPost = renderer.create(<BlogPost cardData = {cardData} />).toJSON();
    expect(blogPost).toMatchSnapshot();
  });
})