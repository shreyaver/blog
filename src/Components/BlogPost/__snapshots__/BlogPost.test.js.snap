// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogPost renders without crashing 1`] = `
<div
  className="Blog-post"
>
  <div
    className="Post-content"
  >
    <div
      className="Image-panel"
      style={
        Object {
          "backgroundImage": "url(abstract.png)",
        }
      }
    />
    <article
      className="Post-text"
    >
      <div>
        <span
          className="Date-posted"
        >
          2nd Januray, 2018
        </span>
        <span
          className="Reading-time"
        >
          2 mins
        </span>
      </div>
      <label>
        The future of abstract art and the culture ...
      </label>
      <p>
        Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...
      </p>
    </article>
  </div>
  <div
    className="Post-reaction"
  >
    <hr />
    <span>
      <button
        className="Clap-icon"
      />
      10
      <button
        className="Heart-icon"
      />
    </span>
  </div>
</div>
`;
